res/enkf/local_ministep.py

```diff
@@ -16,42 +16,58 @@ class LocalMinistep(BaseCClass):
     _get_local_obs_data = ResPrototype(
         "local_obsdata_ref local_ministep_get_obsdata(local_ministep)"
     )
-    _get_local_data = ResPrototype(
-        "local_dataset_ref local_ministep_get_dataset(local_ministep , char*)"
-    )
     _get_obs_data = ResPrototype(
         "obs_data_ref local_ministep_get_obs_data( local_ministep )"
     )
-    _has_local_data = ResPrototype(
-        "bool              local_ministep_has_dataset(local_ministep , char*)"
-    )
     _free = ResPrototype("void local_ministep_free(local_ministep)")
     _attach_obsdata = ResPrototype(
         "void local_ministep_add_obsdata(local_ministep, local_obsdata)"
     )
-    _attach_dataset = ResPrototype(
-        "void local_ministep_add_dataset(local_ministep, local_dataset)"
-    )
     _name = ResPrototype("char* local_ministep_get_name(local_ministep)")
-    _data_size = ResPrototype("int local_ministep_get_num_dataset(local_ministep)")
+    _data_size = ResPrototype("int local_ministep_num_active_data(local_ministep)")
+    _active_data_list = ResPrototype(
+        "active_list_ref local_ministep_get_active_data_list(local_ministep, char*)"
+    )
+    _has_active_data = ResPrototype(
+        "bool local_ministep_data_is_active(local_ministep, char*)"
+    )
+    _add_active_data = ResPrototype(
+        "void local_ministep_activate_data(local_ministep, char*)"
+    )
+    _get_or_create_row_scaling = ResPrototype(
+        "row_scaling_ref local_ministep_get_or_create_row_scaling(local_ministep, char*)"
+    )
 
     def __init__(self, ministep_key):
         raise NotImplementedError("Class can not be instantiated directly!")
 
-    # Will used the data keys; and ignore observation keys.
-    def __getitem__(self, data_key):
-        if isinstance(data_key, int):
-            raise TypeError("Keys must be strings, not int!")
-        if data_key in self:
-            return self._get_local_data(data_key)
+    def initEnsembleConfig(self, config):
```

Comment:
this is not an init function, but a set function.

Reply:
> this is not an init function, but a set function.

Yeah...  it's used exclusively during init, but I see your point...